boss-ws/src/main/java/org/genomebridge/boss/http/service/DatabaseBossAPI.java

```diff
@@ -280,6 +280,28 @@ public ErrorDesc resolveObjectForCopying(String objectId, String userName, CopyR
         return null;
     }
 
+    @Override
+    public ErrorDesc getResumableUploadURL(String objectId, String userName, CopyResponse resp) {
+        ErrorDesc error = null;
+        if ( userName == null  ){
+            return badReqErr(getMessage("remoteUser"));
+        }
+        BossDAO dao = getDao();
+        ObjectRow rec = dao.findObjectById(objectId);
+        if ( rec == null ) {
+            return  notFoundErr(objectId);
+        }
+        if (!rec.storagePlatform.equals(StoragePlatform.CLOUDSTORE.getValue()))
```

Comment:
This needs to be:
if ( rec.storagePlatform.equals(StoragePlatform.OPAQUEURI.getValue()) )

The reason is that the actual object store type (GCS, S3, or FCS) is determined in the config file.  It's independent of the choice of the StoragePlatform begin CLOUDSTORE or LOCALSTORE.  However, OPAQUEURI is unique:  it's not associated with any type of object store.  There's no place to specify that in the config file.

That's why we have to single out and reject that one possibility explicitly.